library/core/src/iter/adapters/flatten.rs

```diff
@@ -444,3 +473,21 @@ where
         init
     }
 }
+
+trait ConstSizeIterable {
+    fn size() -> Option<usize>;
```

Comment:
Ideally this would be a constant but currently specialization doesn't work for associated constants.